kale/embed/factorization.py

```diff
@@ -267,80 +269,123 @@ def inverse_transform(self, x):
+        fit_label (bool): Whether to fit the label for semi-supervised MIDA. Defaults to False.
         kernel_params (dict or None): Parameters for the kernel. Defaults to None.
 
     References:
         [1] Yan, K., Kou, L. and Zhang, D., 2018. Learning domain-invariant subspace using domain features and
             independence maximization. IEEE transactions on cybernetics, 48(1), pp.288-299.
     """
 
+    _parameter_constraints: dict = {
+        "n_components": [
+            Interval(Integral, 1, None, closed="left"),
+            None,
+        ],
+        "kernel": [
+            StrOptions({"linear", "poly", "rbf", "sigmoid", "cosine", "precomputed"}),
+            callable,
+        ],
+        "gamma": [
+            Interval(Real, 0, None, closed="left"),
+            None,
+        ],
+        "degree": [Interval(Real, 0, None, closed="left")],
+        "coef0": [Interval(Real, None, None, closed="neither")],
+        "kernel_params": [dict, None],
+        "mu": [Real],
+        "eta": [Real],
+        "fit_label": ["boolean"],
+        "augmentation": ["boolean"],
+        "n_jobs": [None, Integral],
+    }
+
     def __init__(
         self,
-        n_components,
+        n_components=None,
         kernel="linear",
-        lambda_=1.0,
         mu=1.0,
         eta=1.0,
+        gamma=None,
+        degree=3,
+        coef0=1,
+        fit_label=False,
         augmentation=False,
         kernel_params=None,
+        n_jobs=None,
     ):
         self.n_components = n_components
         self.kernel = kernel
+        self.gamma = gamma
+        self.degree = degree
+        self.coef0 = coef0
         self.mu = mu
         self.eta = eta
         self.augmentation = augmentation
-        if kernel_params is None:
-            self.kernel_params = {}
-        else:
-            self.kernel_params = kernel_params
-        self._label_binarizer = LabelBinarizer(pos_label=1, neg_label=-1)
-        self._centerer = KernelCenterer()
+        self.kernel_params = kernel_params
+        self.fit_label = fit_label
         self.x_fit = None
+        self._label_binarizer = None
+        self._centerer = None
+        self._onehot = None
+        self.n_jobs = n_jobs
 
     def _get_kernel(self, x, y=None):
-        if self.kernel in ["linear", "rbf", "poly"]:
+        if callable(self.kernel):
             params = self.kernel_params or {}
         else:
-            raise ValueError("Pre-computed kernel not supported")
-        return pairwise_kernels(x, y, metric=self.kernel, filter_params=True, **params)
+            self.gamma = 1 / x.shape[1] if self.gamma is None else self.gamma
+            params = {"gamma": self.gamma, "degree": self.degree, "coef0": self.coef0}
+        return pairwise_kernels(x, y, metric=self.kernel, filter_params=True, n_jobs=self.n_jobs, **params)
 
-    def fit(self, x, y=None, covariates=None):
+    def fit(self, x, y=None, groups=None):
```

Comment:
To make `_parameter_constraints` work, don't forget to add `@_fit_context(prefer_skip_nested_validation=True)` on top of `fit` as shown by various `sklearn` estimator's source code.

Reply:
the function could be imported from `sklearn.base`